keyboards/peej/tripel/tripel.h

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2021 Paul James
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define LEFT_PCB 1
+#define MIDDLE_PCB 2
+#define RIGHT_PCB 3
+
+#ifndef PRO_MICRO_POSITION
+    #define PRO_MICRO_POSITION LEFT_PCB
+#endif
+
+#include "quantum.h"
+
+#if (PRO_MICRO_POSITION == LEFT_PCB)
```

Comment:
This stuff won't work in the Configurator.

Reply:
If the configuration info file matches the default here, then it'll work unless you change this value.

Is it better to not have a config info file?

Reply:
That's not what I mean.
https://config.qmk.fm/
How do you expect users to change the value of this define in the Configurator?

Reply:
Well obviously they can't change the define from the configurator. I'm not well versed in the configurator and how it works, is there a way for it to support different matrix configurations for a board? I'm not sure I understand your point.

Reply:
> is there a way for it to support different matrix configurations for a board?

With different revisions/subfolders for the board.

But basically the Configurator doesn't expose any way to change defines, or make options, so it's compiled without any keymap configurations.  

Using a revision would allow you to specify the differences between the hardware layouts/configurations

Reply:
That makes sense. I've never used revisions, I'll see if I can figure them out.

Reply:
A good reference point may be the GMMK Pro being split into ISO and ANSI layouts.
https://github.com/qmk/qmk_firmware/commit/b994921703190886714a4b2df2a674265e586b59